packages/playwright-core/src/server/recorder.ts

```diff
@@ -148,6 +149,31 @@ export class Recorder implements InstrumentationListener, IRecorder {
       this._context.instrumentation.removeListener(this);
       this._recorderApp?.close().catch(() => {});
     });
+
```

Comment:
I would probably extend `_updateUserSources()` to push primary page url to keep things simple.